presto-main/src/main/java/io/prestosql/dispatcher/QueuedStatementResource.java

```diff
@@ -157,6 +157,20 @@ public void stop()
         queryPurger.shutdownNow();
     }
 
+    /**
+     * Gets the information about the query.
+     * <p>
+     * Checks if there is a sql command, takes the remote address from the request, goes
+     * through the headers to instantiate an new object of type {@link HttpRequestSessionContext}
+     * and type {@link Query} and returns a response from the server.
```

Comment:
The method actually receives statement requests which creates new queries on Presto. It doesn't get info about existing queries.